### PR TITLE
Improve delivery report pytests

### PR DESCRIPTION
**Pull request by ivadym**
### Changed
- Removed meta class dependency from the delivery reports
- Updated delivery report pytests
- Added delivery report pytests for all supported workflows
- Some method have been renamed
- Removed mocked delivery report related APIs

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### Tests

| Workflows   | Store Case | Generate Delivery Report | Upload to Scout |
|:-----------:|:--------------:|:--------------------------:|:---------------:|
| raredisease  | X | X | X (manual `delivery-report-to-scout` command) |
| balsamic    | X | X | X |
| mip-dna     | X | X | - |
| rnafusion   | X | X | X |
| taxprofiler  | X | X | - |
| tomte       | X | X | - |

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/meta/delivery_report/test_data_validators.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    assert "duplicates" in missing_fields["metadata"][sample_id]
+    assert "bait_set" not in missing_fields["metadata"][sample_id]
+
+
+def test_get_million_read_pairs():
+    """Test millions read pairs calculation."""
+
+    # GIVEN a number of sequencing reads and its representation in millions of read pairs
+    sample_reads = 1_200_000_000
+    expected_million_read_pairs = 600.0
```
Comment (henrikstranneheim) [inline tests/meta/delivery_report/test_data_validators.py:112]:
```suggestion
    expected_million_read_pairs = sample_reads / 2_000_000
```

File: tests/models/delivery_report/test_models_validators.py [hunk 2]
```diff
@@ -207,9 +221,9 @@ def test_get_analysis_type_as_string():
     """Test analysis type formatting for the delivery report generation."""
 
     # GIVEN a WGS analysis type and a model info dictionary
-    analysis_type: AnalysisType = AnalysisType.WHOLE_GENOME_SEQUENCING
-    model_info: ValidationInfo = ValidationInfo
-    model_info.data: dict[str, Any] = {"workflow": Workflow.MIP_DNA.value}
+    analysis_type = AnalysisType.WHOLE_GENOME_SEQUENCING
```
Comment (henrikstranneheim) [inline tests/models/delivery_report/test_models_validators.py:224]:
```suggestion
    analysis_type: str = AnalysisType.WHOLE_GENOME_SEQUENCING
```

**Review by henrikstranneheim — APPROVED** [review]
Massive work, beautiful, 🥇

File: tests/conftest.py [hunk 3]
```diff
@@ -4092,3 +4112,32 @@ def analysis_summary(
         delivered=delivered_status_summary,
         failed=failed_status_summary,
     )
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (ivadym) [reply tests/conftest.py:4117]: Definitely! Would be nice to create a mocker for the LIMS api as well. But I will leave it like this until we decide to do something with that API

**Comment by ivadym** [comment]
Thanks @henrikstranneheim for the review ❤️

**Comment by ivadym** [comment]
I've tested the dry-run for a couple of workflows after addressing the reviews, and I would say it's ready to be merged

**Comment by ivadym** [comment]
Would greatly appreciate if someone can merge this on monday @Clinical-Genomics/sysdev @Vince-janv ❤️

**Comment by Vince-janv** [comment]
> Would greatly appreciate if someone can merge this on monday @Clinical-Genomics/sysdev @Vince-janv ❤️

Great work! I'll make sure this is merged 😊

**Comment by Vince-janv** [comment]
Deployed in stage and in production

**Pull request merged**